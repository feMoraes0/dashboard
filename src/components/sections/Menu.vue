<template>
  <section>
    <a href="">
      <img src="../../assets/hearthstone-battlegrounds.png" alt="" />
    </a>
    <Button type="button">
      <PlusIcon />
      Add new entry
    </Button>
    <div class="links">
      <Button type="link">
        <HomeIcon />
        Home
      </Button>
      <Button type="link">
        <ChartIcon />
        Charts
      </Button>
      <Button type="link">
        <ListIcon />
        Registers
      </Button>
      <Button type="link">
        <PeopleIcon />
        Heroes
      </Button>
      <Button type="link">
        <SettingsIcon />
        Settings
      </Button>
      <Button type="link">
        <LogoutIcon />
        Sign out
      </Button>
    </div>
  </section>
</template>

<script>
import Button from "../elements/Button.vue";
import HomeIcon from "../icons/Home.vue";
import ChartIcon from "../icons/Chart.vue";
import ListIcon from "../icons/List.vue";
import PeopleIcon from "../icons/People.vue";
import SettingsIcon from "../icons/Settings.vue";
import LogoutIcon from "../icons/Logout.vue";
import PlusIcon from "../icons/Plus.vue";

export default {
  name: "Menu",
  components: {
    Button,
    HomeIcon,
    ChartIcon,
    ListIcon,
    PeopleIcon,
    SettingsIcon,
    LogoutIcon,
    PlusIcon,
  },
};
</script>

<style scoped>
section {
  position: relative;
  height: 100%;
  width: var(--menu-width);
  background-color: var(--white-colour);
  padding-top: 20px;
  padding-left: 30px;
  padding-right: 30px;
}

section img {
  width: 80%;
  margin: 0px 10% 40px;
}

section .links {
  margin-top: 40px;
}
section .links button {
  margin-bottom: 10px;
}
</style>
