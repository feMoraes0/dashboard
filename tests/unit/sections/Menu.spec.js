import { shallowMount } from '@vue/test-utils';
import Menu from '../../../src/components/sections/Menu.vue';

let wrapper;

describe('Menu component', () => {

  it('has Button import', () => {
    // GIVEN
    const { components } = Menu;
    // THEN
    expect(components).toHaveProperty('Button');
  });

  describe('when component is mounted', () => {
    beforeEach(() => {
      wrapper = shallowMount(Menu);
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it('should render to a snapshot when component is initialized', () => {
      //THEN
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});