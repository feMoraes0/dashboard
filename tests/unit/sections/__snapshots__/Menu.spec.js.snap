// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component when component is mounted should render to a snapshot when component is initialized 1`] = `
<section>
  <a
    href=""
  >
    <img
      alt=""
      src="../../assets/hearthstone-battlegrounds.png"
    />
  </a>
   
  <button-stub
    type="button"
  >
    <plusicon-stub />
    
    Add new entry
  
  </button-stub>
   
  <div
    class="links"
  >
    <button-stub
      type="link"
    >
      <homeicon-stub />
      
      Home
    
    </button-stub>
     
    <button-stub
      type="link"
    >
      <charticon-stub />
      
      Charts
    
    </button-stub>
     
    <button-stub
      type="link"
    >
      <listicon-stub />
      
      Registers
    
    </button-stub>
     
    <button-stub
      type="link"
    >
      <peopleicon-stub />
      
      Heroes
    
    </button-stub>
     
    <button-stub
      type="link"
    >
      <settingsicon-stub />
      
      Settings
    
    </button-stub>
     
    <button-stub
      type="link"
    >
      <logouticon-stub />
      
      Sign out
    
    </button-stub>
  </div>
</section>
`;
